tiledb/sm/query/readers/reader_base.cc

```diff
@@ -1253,6 +1293,44 @@ Status ReaderBase::unfilter_tile_chunk_range(
   return Status::Ok();
 }
 
+Status ReaderBase::unfilter_tile_chunk_range(
+      uint64_t num_range_threads,
+      uint64_t thread_idx,
+      const std::string& name,
+      Tile* tile,
+      const ChunkData& tile_chunk_data,
+      BitSortFilterMetadataType &pair) const {
+  assert(tile);
+  // Prevent processing past the end of chunks in case there are more
+  // threads than chunks.
+  if (thread_idx > tile_chunk_data.filtered_chunks_.size() - 1) {
+    return Status::Ok();
+  }
+
+  FilterPipeline filters = array_schema_.filters(name);
+
+  // Append an encryption unfilter when necessary.
+  RETURN_NOT_OK(FilterPipeline::append_encryption_filter(
+      &filters, array_->get_encryption_key()));
+
+  // Compute chunk boundaries
+  auto&& [t_min, t_max] = compute_chunk_min_max(
+      tile_chunk_data.chunk_offsets_.size(), num_range_threads, thread_idx);
+
+  // Reverse the tile filters.
```

Comment:
I'm not sure I understand here but this is run_reverse_chunk_range for HasBitSortFilter = true always?

Reply:
Yes. Well we set the template parameter to true when there is a bitsort filter. Once again this is to ensure we don't have a bunch of runtime checks throughout the code.

Reply:
Oh I see, just to confirm I understand this correctly, this function (ReaderBase::unfilter_tile_chunk_range) will only get called when we have a bitsort filter right?

Reply:
Yes!